src/ray/telemetry/ray_event_recorder_base.cc

```diff
@@ -21,11 +21,24 @@ namespace telemetry {
 
 template <typename TEventData>
 RayEventRecorderBase<TEventData>::RayEventRecorderBase(
-    rpc::EventAggregatorClient &event_aggregator_client,
+    std::unique_ptr<rpc::EventAggregatorClient> event_aggregator_client,
     instrumented_io_context &io_service)
-    : event_aggregator_client_(event_aggregator_client),
+    : event_aggregator_client_(std::move(event_aggregator_client)),
       periodical_runner_(PeriodicalRunner::Create(io_service)) {}
 
+template <typename TEventData>
+RayEventRecorderBase<TEventData>::RayEventRecorderBase(
```

Comment:
why do we need a base class for this? my understanding is the event data is all of type `rpc::events::RayEventsData`

so we should be able to use a single event export client